File: 442-Find-All-Duplicates-in-an-Array/ts/main.ts
function findDuplicates(nums: number[]): number[] {
    const duplicates: number[] = [];
    nums.forEach((num) => {
        const index = Math.abs(num) - 1;
        if (nums[index] < 0) {
            duplicates.push(Math.abs(num));
        } else {
            nums[index] = -nums[index];
        }
    });
    return duplicates;
}

File: 442-Find-All-Duplicates-in-an-Array/go/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println("Hello World")
	nums := []int{
		4, 3, 2, 7, 8, 2, 3, 1,
	}
	fmt.Println("duplicates =>", findDuplicates(nums))
}

func findDuplicates(nums []int) []int {
	var duplicates []int

	for _, num := range nums {
		index := int(math.Abs(float64(num))) - 1

		if nums[index] < 0 {
			duplicates = append(duplicates, int(math.Abs(float64(num))))
		} else {
			nums[index] = -nums[index]
		}
	}
	return duplicates
}
